avocado/core/sysinfo.py

```diff
@@ -213,6 +214,45 @@ def stop(self):
         return retcode
 
 
+class JournalctlWatcher(Collectible):
+
+    """
+    Track the content of systemd journal into a compressed file.
+
+    :param logf: Basename of the file where output is logged (optional).
+    """
+
+    def __init__(self, logf=None):
+        if not logf:
+            logf = 'journalctl.gz'
+
+        super(JournalctlWatcher, self).__init__(logf)
+        self.cursor = self._get_cursor()
+
+    def _get_cursor(self):
+        try:
+            cmd = 'journalctl --lines 1 --output json'
+            result = process.system_output(cmd, verbose=False)
+            last_record = json.loads(result)
+            return last_record['__CURSOR']
+        except Exception as e:
+            log.debug("Journalctl collection failed: %s", e)
+
+    def run(self, logdir):
+        if self.cursor:
```

Comment:
Note to self: gets implicitly set to `None` if exception is raised and `_get_cursor()` returns `None`.